[Installer] Possibility to skip database connection setup (database.yml)

# Bug Report

I am trying to update the installer for Pimcore on Platform.sh, and running into an issue.  Specifically, even though an `/appconfig/installer.yml` file is included, the installer is tryign to write an `/app/config/local/database.yml` file.  I want it to skip that entirely, because the DB information will always be provided via environment variables and injected into the container through a custom parameters file.

I don't believe this problem existed on earlier versions, but with the latest Pimcore release I'm running into this wall and haven't figured out a way around it.

### Expected/Desired behavior

Installer detects that DB information is already available in the container and just keeps on going without trying to write any YAML files to disk.

### Actual behavior

The installer fails with:

```
 [ERROR] The following errors were encountered during installation                                                      
                                                                                                                        

 * Warning: file_put_contents(/app/app/config/local/database.yml): failed to open stream: Read-only file system
```

Unfortunately I think this hard-blocks the Pimcore template on Platform.sh for the time being.

--- a/bundles/InstallBundle/Command/InstallCommand.php
+++ b/bundles/InstallBundle/Command/InstallCommand.php
@@ -177,6 +177,11 @@ class InstallCommand extends Command
                 null,
                 InputOption::VALUE_NONE,
                 'Do not abort if a <comment>system.yml</comment> file already exists'
+            )->addOption(
+                'skip-database-config',
+                null,
+                InputOption::VALUE_NONE,
+                'Do not write a database config file: <comment>database.yml</comment>'
             );
 
         foreach ($this->getOptions() as $name => $config) {
@@ -198,6 +203,10 @@ class InstallCommand extends Command
         $configFile = Config::locateConfigFile('system.yml');
         if ($configFile && is_file($configFile) && !$input->getOption('ignore-existing-config')) {
             throw new \RuntimeException(sprintf('The system.yml config file already exists in "%s". You can run this command with the --ignore-existing-config flag to ignore this error.', $configFile));
+        }
+
+        if ($input->getOption('skip-database-config')) {
+            $this->installer->setSkipDatabaseConfig(true);
         }
 
         $this->io = new PimcoreStyle($input, $output);

--- a/bundles/InstallBundle/Installer.php
+++ b/bundles/InstallBundle/Installer.php
@@ -88,6 +88,21 @@ class Installer
     private $importDatabaseDataDump = true;
 
     /**
+     * skip writing database.yml file
+     *
+     * @var bool
+     */
+    private $skipDatabaseConfig = false;
+
+    /**
+     * @param bool $skipDatabaseConfig
+     */
+    public function setSkipDatabaseConfig(bool $skipDatabaseConfig): void
+    {
+        $this->skipDatabaseConfig = $skipDatabaseConfig;
+    }
+
+    /**
      * @var array
      */
     private $stepEvents = [
@@ -542,7 +557,11 @@ class Installer
     private function createConfigFiles(array $config)
     {
         $writer = new ConfigWriter();
-        $writer->writeDbConfig($config);
+
+        if (!$this->skipDatabaseConfig) {
+            $writer->writeDbConfig($config);
+        }
+
         $writer->writeSystemConfig();
         $writer->writeDebugModeConfig();
         $writer->generateParametersFile();

